taichi/analysis/same_statements.cpp

```diff
@@ -13,23 +14,35 @@ class IRNodeComparator : public IRVisitor {
   IRNode *other_node;
   // map the id from this node to the other node
   std::unordered_map<int, int> id_map;
+
   // ids which don't belong to either node
   std::unordered_set<int> captured_id;
```

Comment:
A bit confused by the word `captured` - what does `captured_id` mean here?

Reply:
I chose the word "captured" because it's similar to lambda expression's "capture".

In this case (calling `same_statements($3, $6)`), `$1` and `$2` should be "captured" since they don't belong to either `$3` or `$6`.
```
<i32> $3 = add $1 $2
<i32> $6 = add $1 $2
```
Here "belong" means that "in the AST of root1/root2". A larger example when `$1` and `$2` are "captured":
```
$3: range_for($1, $2) {
  $4 = loop $3 index 0
  $5 = const [1]
}
$6: range_for($1, $2) {
  $7 = loop $6 index 0
  $8 = const [1]
}
```